<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>Lab meeting 09.04.2018</title>

		<meta name="description" content="Lab meeting 09.04.2018<">
		<meta name="author" content="Aaron Brooks">

		<link rel="stylesheet" href="css/reveal.css">
		<link rel="stylesheet" href="css/theme/embl.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>

		<style>
		.rotate {
	    -webkit-transform: rotate(25deg);
	    -moz-transform: rotate(25deg);
	    -o-transform: rotate(25deg);
	    -ms-transform: rotate(25deg);
	    transform: rotate(25deg);
		}
		</style>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">
				<section data-background-color="#2A292D" id="slide1">
					<h2 style="color:white">Transcription in a <b><span
						style="color:#9932CC">S</span><span
						style="color:#4169E1">C</span><span
						style="color:#3CB371">R</span></b>a<b><span
						style="color:#FFFF33">M</span></b>b<b><span
						style="color:#FF8C00">L</span><span
						style="color:#FF0000">E</span></b>d genome</h2>
					<p style="color:white">
 						<a href="http://aaronbrooks.info">Aaron Brooks</a> /
						<img height=40 align="middle"
							style="background:0px;margin:0px 5px 5px 5px;border:0;box-shadow:0;vertical-align:middle"
							data-src="../../src/TwitterLogo_white.png"><a href="http://twitter.com/scalefreegan">@scalefreegan</a>
					</p>
					<p style="color:white">Steinmetz and Stegle Groups, EMBL Heidelberg</p>
					<p style="color:white">
						<small>Follow along on  <img height=40 align="middle" style="background:0px;margin:0px 5px 5px 5px;border:0;box-shadow:0;vertical-align:middle" data-src="../../src/GitHub-Mark/PNG/GitHub-Mark-Light-64px.png"></small>
						<br>
						<small><a href="http://bit.ly/2tGogze">bit.ly/2tGogze</a></small>
					</p>
				</section>

				<section id="pre">
					<section id="slide2">
							<h3>...in the not too distant future</h3>
							<img width="45%" src="../../src/buildgenome.gif" style="border:0px;box-shadow:0 0 0;margin:0 0"></img>
							<p>you are building a genome from scratch...</p>
							<p><small><span class="fragment">Where are you going to put the genes?</span> <span class="fragment">In what order and which orientation?</span> <span class="fragment">Does it matter?</span></small></p>
							<p><small>
								<a
								href="http://science.sciencemag.org/content/343/6178/1426.full">
								Science Magazine, Mar 28th 2014</a>
							</small></p>
					</section>
					<section id="slide3">
							<h3 >Transcription is linked to genome architecture</h3>
							<img width="75%" src="../../src/TIFSeq-1024x650.jpg" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small>
								<a href="http://pelechanolab.com/">Pelechano Lab</a>
							</small></p>
					</section>
					<section id="slide4">
							<h3 >Regulatory signals can propagate between neighbouring promoters</h3>
							<img width="65%" src="../../src/nrg3594-f4_mod-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small>
								<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3063692/">Xu et al 2011</a><br>
								<a href="http://www.nature.com/nrg/journal/v14/n12/full/nrg3594.html">Pelechano and Steinmetz 2013</a>
							</small></p>
							<aside class="notes">
								Antisense transcripts can transmit regulatory signals to neighbouring promoters29.
								Upon galactose induction in budding yeast, the same transcription factor (TF)-mediated
								activation pathway induces the expression of GAL80 and an antisense transcript that
								originates bidirectionally from its promoter. This antisense transcript runs upstream
								into the promoter of SUR7 and represses its activity, thereby spreading GAL80 regulatory
								signal to both genes
							</aside>
					</section>
					<section id="slide5">
							<h3 >Antisense-mediated regulation of SUR7</h3>
							<img width="100%" src="../../src/msb20111-f4-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small>
								<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3063692/">Xu et al 2011</a><br>
							</small></p>
							<aside class="notes">
								(C) Mean of the log2 expression levels of SUR7 across three biological replicates for strains under α-factor stimulation grown in YPD (purple) or YPGal (turquoise), for the wild-type (left) or the antisense-disrupted strain (right). Values of individual samples are shown by open circles. (D) Model for spreading of regulatory signals on the SUR7–GAL80 locus. SUR7 and GAL80 are two ORFs (blue boxes) in tandem configuration. The promoter of GAL80 is bidirectional and initiates SUT719, a transcript antisense to SUR7. Regulatory signals in response to galactose media (Gal4 transcription factor, green oval) upregulate expression of GAL80 as well as the antisense SUT719, thereby repressing SUR7.
							</aside>
					</section>
					<section id="slide6">
							<h3 >Conservation of organization despite regulatory divergence</h3>
							<img width="75%" src="../../src/gal1_gal10.jpg" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small>
								<a href="https://www.sciencedirect.com/science/article/pii/S0960982207013954#fig1/">Martchenko et al 2011</a><br>
							</small></p>
							<aside class="notes">
								Figure 1. Analysis of the GAL Circuit across Ascomycota

								(A) The structural members of the GAL regulon are conserved in Ascomycota and are also organized in a cluster. Species in which whole-genome duplication occurred are identified by the star and by the color. The regions of D. hansenii, C. parapsylopsis, C. dubliniensis, and C. albicans also contain two uncharacterized open reading frames X and Y, marked in black and white color; S. pombe contains only one of these elements. The primary sequences of the GAL promoters were analyzed for the presence of Gal4-, Cph1-, and Mig1-binding sites by the motif-finding program MEME (http://weblogo.berkley.edu/). The phylogenetic relationships among the fungi are shown as defined in [45].

								(B) ScGal4p acts as the activator of the S. cerevisiae GAL regulon. The putative CaGal4p protein, encoded by C. albicans ORF19.5338, shares strong sequence similarity (86%) with S. cerevisiae Gal4p only within the DNA-binding domain; the DNA-binding domain has the six cysteine residues, the linker region, and the dimerization region all well conserved.
							</aside>
					</section>
					<!-- <section id="slide6">
							<h3 >Functional consequences of bidirectional promoters</h3>
							<img width="100%" src="../../src/nihms298426f3_2.jpg" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small>
								<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3123404/">Wei et al 2012</a><br>
							</small></p>
							<aside class="notes">
								Non-coding RNA transcription from bidirectional promoters couples the expression of protein-coding genes through local effects (A and B), regional signal spreading (C), or distal effects (D). A) Transcription of ncRNAs reorganizes (green arrows) the chromatin in bidirectional regions or competes (dark blue line) with the protein-coding gene for the same pool of polymerases and accessory factors. Transcription of ncRNAs can modify (left green arrow) the local chromatin structure, for example by displacing positioned nucleosomes (transparent one), therefore facilitating (right green arrow) the transcription of the protein-coding gene. B) ncRNA transcripts affect histone modifications of nearby chromatin. The solid green line shows the repression effect published in case studies [17, 77]; dashed green lines show possible effects suggested in this review. C) Transcription factors co-regulate the expression of a ncRNA (blue arrow) and Gene 2. The ncRNA then spreads the signal to the overlapping Gene 1 (green line) and represses its expression (red arrows) as demonstrated recently [78]. D) ncRNAs affect activation or repression (green arrows) of distal genes (gene in dashed oval). The bidirectional promoter can generate only ncRNAs or a protein-coding gene and a ncRNA (not shown). The direction of transcription is represented for non-coding transcription (blue arrow) and protein-coding transcription (red arrow). Protein-coding regions are represented by dark blue boxes.
							</aside>
					</section> -->
					<section id="slide7">
							<blockquote style="box-shadow:0 0 0"cite="http://www.nature.com/nrg/journal/v14/n12/full/nrg3594.html">
								&ldquo;...gene shuffling could be evolutionarily restricted in regions that support
								beneficial interactions between loci, whereas in other regions, <b>new rearrangements
								could give rise to novel interaction networks that are mediated by ncRNAs and
								antisense transcripts.</b>&rdquo;
							</blockquote>
							<p><small>
								<a href="http://www.nature.com/nrg/journal/v14/n12/full/nrg3594.html">Pelechano and Steinmetz 2013</a>
							</small></p>
					</section>
					<section id="slide8">
							<h3 >Synteny is conserved among closely related yeast species</h3>
							<img width="100%" src="../../src/kellis_2003_fig1.jpg" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small>
								<a href="http://www.nature.com/nature/journal/v423/n6937/full/nature01644.html">Kellis et al 2003</a>
							</small></p>
							<aside class="notes">
								Figure 1: Aligned ORFs across four species. A 50-kb segment of S. cerevisiae chromosome VII
								aligned with orthologous contigs from each of the other three species.
								Predicted ORFs are shown as arrows pointing in the direction of transcription.
								Orthologous ORFs are connected by dotted lines and are coloured by the type of
								correspondence: red for 1-to-1 matches, blue for 1-to-2 matches and white for
								unmatched ORFs. Sequence gaps are indicated by vertical lines at the ends of contigs,
								with the estimated size of each gap shown by the length of the hook.
								Most ORFs have clear one-to-one orthologous matches in each species, providing a dense set of
								landmarks (average spacing approx2 kb) to define blocks of conserved synteny covering essentially the
								entire genome (Fig. 1). For a small number of ORFs (211), the correspondence is ambiguous, however.
								These ambiguous matches almost all reflect local gene-family expansion or contraction.
								Most of the ambiguities are markedly clustered in telomeric regions (Fig. 2). More than 80% fall into one of 32
								clusters of two or more genes (average size approx18 kb, together comprising approx4% of the genome), which
								correspond nearly perfectly to the 32 telomeric regions of the 16 chromosomes of S. cerevisiae. Only one telomeric
								region lacks a cluster (chromosome VI-R). And only one cluster does not lie in telomeric regions in S. cerevisiae: it
								is a recent insertion of a segment that is telomeric in the other three species.
								All non-telomeric translocations are mediated by Ty elements (2, 4, 5, 6, 7, 8, 9) or nearly identical copies of ribosomal protein genes
							</aside>
					</section>
					<section id="slide9">
							<h3 >Synteny decreases with evolutionary distance</h3>
							<img width="100%" src="../../src/dujon_2003_mod-01.jpg" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small><i>&ldquo;...different yeast lineages have evolved through a marked interplay between several distinct molecular mechanisms,
								including tandem gene repeat formation, segmental duplication, a massive genome duplication and extensive gene loss.&rdquo;</i></small></p>
							<p><small>
								<a href="http://www.nature.com/nature/journal/v430/n6995/full/nature02579.html#B14">Dujon et al 2004</a>
							</small></p>
							<aside class="notes">
								Chromosomal blocks containing at least three pairs of paralogous genes with a probability
								lower than 0.01 to occur by chance (see Methods) are plotted on two-dimensional diagrams
								each representing an entire yeast genome (chromosomes, represented by shifted bars, are
								ordered by increasing numbers from the lower-left corner). Each cross on the plots
								corresponds to a pair of paralogous genes belonging to a defined block. The shaded
								triangle in each plot gives numerical descriptions of the duplicated blocks found for
								each species. Species abbreviations as in Fig. 1.
							</aside>
					</section>
					<section id="slide10">
							<h3>Local genome organization is non-random</h3>
							<p><i><em><u>~50%</u></em> of orthologous genes in human and fish belong to conserved synteny blocks</i></p>
							<p><small>
								<a href="https://www.ncbi.nlm.nih.gov/pubmed/18929678">Koonin 2009</a>
							</small></p>
							<img width="55%" src="../../src/harvey_bgc.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small>
								<a href="https://www.biorxiv.org/content/early/2018/01/15/247940">Harvery et al 2018</a>
							</small></p>
					</section>
				</section>

				<section id="intro">
					<section
						data-background="../../src/Yeast-Cell.jpg"
						data-background-size="contain"
						data-background-color="#2A292D"
						id="slide11">
						<h3 style="color:#EEE;background-color:#007C80;border-radius: 15px;"> &nbsp;Why are genomes organized as we observe them?&nbsp; </h3>
						<!-- <h3 class="fragment" style="color:#EEE;background-color:#007C80;border-radius: 15px;"> &nbsp;What happens when we rearrange them?&nbsp; </h3> -->
					</section>
					<section id="slide12">
							<h3><i>Gene shuffling</i> with SCRaMbLE</h3>
							<p><small><u>S</u>ynthetic <u>C</u>hromosome <u>R</u>ecombination <u>a</u>nd <u>M</u>odification <u>b</u>y <u>L</u>oxP-mediated <u>E</u>volution</small></p>
							<img width="100%" src="../../src/scramble-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small>
								<a href="http://www.tandfonline.com/doi/abs/10.4161/bbug.19543">Dymond and Boeke 2012</a>
							</small></p>
					</section>
					<section id="slide13">
							<h3>"A big step towards an artificial yeast genome"</h3>
							<img width="100%" src="../../src/20170311_STD001_0.jpg" style="border:0px;box-shadow:0 0 0;margin:0 0"></img>
							<p><small>
								<a
								href="http://www.economist.com/news/science-and-technology/21718479-success-would-usher-true-genetic-engineering-big-step-towards-artificial">
								Economist, Mar 11th 2017</a>
							</small></p>
					</section>
					<section id="slide14">
							<h3>"Remodeling the yeast genome piece by piece"</h3>
							<img width="100%" src="../../src/sc20_science.png" style="border:0px;box-shadow:0 0 0;margin:0 0"></img>
							<p><small>
								<a href="http://science.sciencemag.org/content/355/6329/1038">Synthetic Yeast Genome Project (Sc2.0) Science Special Issue</a>
							</small></p>
					</section>
					<section id="slide15">
						<h3 >Encode sequence diversity by design: <b><u>Sc2.0</u></b></h3>
							<img width="60%" src="../../src/sc2_redesign-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small>
								<a href="http://science.sciencemag.org/content/343/6178/1426.full">Pennisi et al. 2014</a>
							</small></p>
					</section>
					<section id="slide16">
							<h3><i>Gene shuffling</i> with SCRaMbLE</h3>
							<p><small><u>S</u>ynthetic <u>C</u>hromosome <u>R</u>ecombination <u>a</u>nd <u>M</u>odification <u>b</u>y <u>L</u>oxP-mediated <u>E</u>volution</small></p>
							<img width="90%" src="../../src/kbie_a_10919543_f0002.jpeg" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small>
								<a href="http://www.tandfonline.com/doi/abs/10.4161/bbug.19543">Dymond and Boeke 2012</a>
							</small></p>
					</section>
					<section id="slide17">
							<h4 >Extensive heterogeneity among SCRaMbLE strains</h4>
							<img width="60%" src="../../src/scramble.jpg" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small>
								<a href="http://genome.cshlp.org/content/26/1/36.long">Shen et al. 2016</a>
							</small></p>
					</section>
					<section id="slide18">
							<h3>How is the transcriptional landscape modified by genome SCRaMbLEing?</h3>
							<img width="100%" src="../../src/scramble_slide-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>

				</section>

				<section id="rnaseq">
					<section
						data-background="../../src/genome-yeast-846x410-01.png"
						data-background-size="contain"
						data-background-color="#2A292D"
						id="slide19">
						<h3 style="color:#EEE;background-color:#007C80;border-radius: 15px;"> &nbsp;How is a SCRaMbLEd genome transcribed?&nbsp;</h3>
						<img width="15%" src="../../src/andreas-01.jpg" style="float:right;box-shadow:0 0 0;margin:0 10px">
						<span style="color:#FFF;background-color:#2A292D;position:absolute;right:20px;bottom:-35px;border-radius: 15px;"> &nbsp;Andreas&nbsp; </span></p>
						<img width="10%" src="../../src/sandra.jpg" style="float:right;box-shadow:0 0 0;margin:0 10px">
						<span style="color:#FFF;background-color:#2A292D;position:absolute;right:175px;bottom:-35px;border-radius: 15px;"> &nbsp;Sandra&nbsp; </span></p>
					</section>
					<section id="slide20">
							<h3 >Sequencing methods to measure changes in the transcriptome</h3>
							<img width="65%" src="../../src/rna_seq_1-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide21">
							<h3 >SynIXR SCRaMbLE Strains</h3>
							<img width="100%" src="../../src/strains-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<!-- <section id="slide22">
							<h3 >3'-end profiles change after SCRaMbLE</h3>
							<img width="90%" src="../../src/seg13-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section> -->
					<section id="slide23">
							<h3 >Mapping 3'-ends is problematic with short reads</h3>
							<img width="100%" src="../../src/seg_multimap-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide24">
							<h3 >Comparing segment similarity with kernel methods</h3>
							<img width="90%" src="../../src/kernel.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide26">
							<h3 >Native Chr 3'-ends do not vary</h3>
							<iframe src="http://localhost/~brooks/plots/chrV_3prime_var.html" width=800px height=500px></iframe>
							<p><small>
								Kernel PCA (Cosine kernel), ChrV
							</small></p>
					</section>
					<section id="slide27">
							<h3 >SCRaMbLEd SynIXR 3'-ends vary</h3>
							<iframe src="http://localhost/~brooks/plots/3prime_var.html" width=800px height=500px></iframe>
							<p><small>
								Kernel PCA (Cosine kernel) from 22 segments represented in all strains
							</small></p>
					</section>
					<section id="slide28">
							<h3 >Sequencing methods to measure changes in the transcriptome</h3>
							<img width="65%" src="../../src/rna_seq_2-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide29">
							<h3 >Long read sequencing technologies</h3>
							<img width="70%" src="../../src/longread_techs-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide30">
							<h3 >Multiple options for full-length RNA prep</h3>
							<img width="100%" src="../../src/fulllen_libprep-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide31">
							<h3 >Are Smart-Seq2 reads not full-length?</h3>
							<img width="60%" src="../../src/smartseq_ercc.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide32">
							<h3 >Qualitatively different results depending on prep</h3>
							<img width="100%" src="../../src/longread_synixr_full-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide33">
							<h3 >synIXR: transcript isoforms that cross loxPsym sites</h3>
							<img width="100%" src="../../src/longread_synixr_zoom-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide34">
							<h3 >Each method has its advantages</h3>
							<img width="100%" src="../../src/directrna_synixr_full-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide35">
							<h3 >Each method has its advantages</h3>
							<img width="100%" src="../../src/directrna_synixr_zoom-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section
						data-background="../../src/nanopore2-01.jpg"
						data-background-size="contain"
						id="slide36">
						<h3 style="color:#EEE;background-color:#007C80;border-radius: 15px;"> &nbsp;Full-length transcript isoform sequencing:<br />global properties&nbsp;</h3>
					</section>
					<section id="slide37">
							<h3 >Small differences in read lengths and quality</h3>
							<img width="100%" src="../../src/longread_lenqual-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide38">
							<h3 >Global genic 5'-end profile</h3>
							<img width="100%" src="../../src/longread_5p-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide39">
							<h3 >Global genic 3'-end profile</h3>
							<img width="100%" src="../../src/longread_3p-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide40">
							<h3 >What fraction of the genome is expressed?</h3>
							<img width="100%" src="../../src/longread_rna_genomecov.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p>"A total of 85% of the genome is expressed in rich media."</p>
							<p><small>
								<a href="http://www.pnas.org/content/103/14/5320">David et al. 2006</a>
							</small></p>
					</section>
					<section id="slide41">
							<h3 >Not many mTIFs are recapitulated</h3>
							<img width="80%" src="../../src/longread_mtif_overlap-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
							<p><small>
								<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3705217/">Pelechano, Wei and Steinmetz 2013</a>
							</small></p>
					</section>
					<section id="slide42">
							<h3 >Different mTIFs re-discovered by each method</h3>
							<img width="60%" src="../../src/longread_mtif_venn-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section
						data-background="../../src/nanopore2-01.jpg"
						data-background-size="contain"
						id="slide43">
						<h3 style="color:#EEE;background-color:#007C80;border-radius: 15px;"> &nbsp;Getting Smart with UMIs:<br />UMIs for Smart-Seq2&nbsp;</h3>
					</section>
					<section id="slide44">
							<h3 >Dual UMIs can account for high error rate with Nanopore</h3>
							<img width="60%" src="../../src/dualumi_smart-01.png" style="border:0px;box-shadow:0 0 0;margin:0 0">
					</section>
					<section id="slide45">
						<h3 style="color:#EEE;background-color:#007C80;border-radius: 15px;"> &nbsp;Summary&nbsp;</h3>
						<ul>
							<li>
								Global 3'-end variation on SynIXR SCRaMbLEd chromosomes, but not native chromosomes
							<br>
							<li class="fragment" style="color:black">
								<b>Direct RNA</b> sequences "everything", is "inherently" strand-specific, no PCR, coolest and makes Sandra happiest (i.e. easiest)
							<br>
							<li class="fragment" style="color:black">
								<b>TeloPrime</b> is specific for capped molecules, but suffers from coverage issues
							<br>
							<li class="fragment" style="color:black">
								<b>Smart-Seq2</b> can amplify degradation products and possibly other non-full-length transcripts, but...
							<br>
							<li class="fragment" style="color:black">
								We can use dual UMIs with Smart-Seq2 to identify PCR duplicates
						</ul>
					</section>
				</section>

				<section id="thanks">
					<section id="slide46"
						data-background="../../src/thanks_2018.png"
						data-background-size="contain">
					</section>
				</section>

			</div>
		</div>
		<div id="myLogo" style="position: absolute;
                        bottom: 25px;
                        left: 45px;
                        width: 125px;
                        height: 75px;">
		<a href="https://www.embl.de/"><img width="50%" src="../../src/embl.png"></img></a>
		</div>
		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.js"></script>

		<script>
			// More info https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				history: true,

				// More info https://github.com/hakimel/reveal.js#dependencies
				dependencies: [
					{ src: 'plugin/markdown/marked.js' },
					{ src: 'plugin/markdown/markdown.js' },
					{ src: 'plugin/notes/notes.js', async: true },
					{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } }
				]
			});
		</script>
	</body>
</html>
